lib/deep-equal.test.js

```diff
@@ -449,29 +452,43 @@ describe("deepEqual", function() {
         assert.isFalse(checkDeep);
     });
 
-    it("returns false if arguments to array", function() {
-        var gather = function() {
-            return arguments;
-        };
-        var checkDeep = samsam.deepEqual([1, 2, {}, []], gather(1, 2, {}, []));
-        assert.isFalse(checkDeep);
-    });
+    if (typeof Symbol === "symbol") {
+        it("returns false if array to arguments", function() {
+            var gather = function() {
+                return arguments;
+            };
+            var checkDeep = samsam.deepEqual(
+                [1, 2, {}, []],
+                gather(1, 2, {}, [])
+            );
+            assert.isFalse(checkDeep);
+        });
+
+        it("returns false if array like object to arguments", function() {
+            var gather = function() {
+                return arguments;
+            };
+            var arrayLike = { length: 4, "0": 1, "1": 2, "2": {}, "3": [] };
+            var checkDeep = samsam.deepEqual(arrayLike, gather(1, 2, {}, []));
```

Comment:
MINOR: Any value in this single-use variable used on the very next line? Could we just do this inside the assertion?

Reply:
I'm just following the style in the rest of the test file here. All other tests create a variable and assert its value. I think the idea is to have the test execution and the assertions on separate lines.

Reply:
Fair enough.